experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1373 @@
+###############################################################################
+#
+#  Construction 
+#
+###############################################################################
+
+##### Algebras #####
+
+### Difference ###
+@doc raw"""
+    difference_polynomial_ring(R::Ring, nelementary_symbols::Int, ndiffs::Int) -> Tuple{DifferencePolyRing, Vector{DifferencePolyRingElem}}
+
+Return a tuple consisting of the difference polynomial ring over the ring `R` with the specified number of elementary variables and commuting endomorphisms, and the vector of
+these elementary variables.This methods allows for all the keywords of the `set_ranking!` method. See there for further information.
+
+# Examples
+
+```jldoctests
+julia> S, variables = difference_polynomial_ring(QQ, 3, 4)
+(Difference polynomial ring in 3 elementary symbols over QQ, DifferencePolyRingElem{QQFieldElem}[u1[0,0,0,0], u2[0,0,0,0], u3[0,0,0,0]])
+
+julia> S
+Difference polynomial ring in 3 elementary symbols u1, u2, u3
+with 4 commuting endomorphisms
+  over Rational field
+
+julia> variables
+3-element Vector{DifferencePolyRingElem{QQFieldElem}}:
+ u1[0,0,0,0]
+ u2[0,0,0,0]
+ u3[0,0,0,0]
+```
+"""
+function difference_polynomial_ring(R::Ring, nelementary_symbols::Int, ndiffs::Int;
+    partition_name::Symbol = :default,
+    index_ordering_name::Symbol = :default,
+    partition::Vector{Vector{Int}} = Vector{Int}[],
+    index_ordering_matrix::ZZMatrix = zero_matrix(ZZ, 0, 0),
+  )
+  dpr = DifferencePolyRing{elem_type(typeof(R))}(R, nelementary_symbols, ndiffs)
+  set_ranking!(dpr;
+      partition_name = partition_name,
+      index_ordering_name = index_ordering_name,
+      partition = partition,
+      index_ordering_matrix = index_ordering_matrix
```

Comment:
```suggestion
    kwargs...
  )
  dpr = DifferencePolyRing{elem_type(typeof(R))}(R, nelementary_symbols, ndiffs)
  set_ranking!(dpr;
      kwargs...
```
It is also possible to just redirect all kwargs. That way you don't have to type them multiple times, and just have a single place (the `set_ranking!`) function that handles the default values.